lib/npg_pipeline/base_resource.pm

```diff
@@ -0,0 +1,204 @@
+package npg_pipeline::base_resource;
+
+use Moose;
+use MooseX::Getopt::Meta::Attribute::Trait::NoGetopt;
+use namespace::autoclean;
+use Readonly;
+use npg_pipeline::function::definition;
+
+Readonly::Scalar my $GB_TO_MB_CONVERSION => 1000; # 1024? Not in original code
+
+our $VERSION = '0';
+
+extends 'npg_pipeline::base';
+
+=head1 NAME
+
+npg_pipeline::base_resource
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+Inheritable class that understands resources as defined in the pipeline JGF
+format. It provides methods for accessing default resources, as well as merged
+resource defintions for special sub-tasks in a pipeline function, e.g.
+particular memory requirements for a certain type of alignment, or a specific
+set of properties for a Portable Pipeline
+
+=head1 SUBROUTINES/METHODS
+=cut
+
+=head2 resource
+
+HashRef of resource requests for the function, e.g.
+{
+  default => {
+    minimum_cpu => 4,
+    maximum_cpu => 8,
+    memory => 10,
+    db => ['mlwh']
+  }
+}
+
+=cut
+
+has resource => (
+  isa => 'HashRef',
+  is => 'ro',
+  default => sub {{}},
+  documentation => 'Function-specific resource spec',
```

Comment:
With `metaclass => 'NoGetopt'` this documentation string will not serve it's purpose of appearing in a help message for a script. Does not harm either.